MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -40,7 +40,10 @@ open class SimulatedLocationManager: NavigationLocationManager {
      Specify the multiplier to use when calculating speed based on the RouteLeg’s `expectedSegmentTravelTimes`.
      */
     @objc public var speedMultiplier: Double = 1
-    
+
+    /// Instead of following the given route, go slightly off route. Useful for testing rerouting.
+    @objc public var goOffRoute: Bool = false
```

Comment:
I used this while testing and thought it might be useful to others. If you disagree, I can revert this portion - no big deal.

Reply:
I think this is a nice idea, I am not sure about the implementation though. Would it be possible to have a delegate or something like that so implementers can decide how the offroute is calculated/offset?

I am open to how this is implemented, but some form of overwriting a method might be a good solution IMHO.